x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -307,7 +353,7 @@ func createMetricDataQuery(metric cloudwatch.Metric, instanceID string, index in
 	for _, dim := range metricDims {
 		if *dim.Name == "InstanceId" && *dim.Value == instanceID {
 			metricName := *metric.MetricName
-			label := instanceID + " " + metricName
+			label := instanceID + labelSeperator + metricName
```

Comment:
Space was used as separator before, what is the motivation to change it to `|`?
Is it a valid character for labels? If it is we may face problems  when splitting the labels if one of them contain the separator character.

Reply:
Yeah this has been bother me. Seems like AWS allows you to put any character for label. `|` is better here as the separator than space but not ideal. It's time to think about other ways of refactoring this.

Reply:
My fault, I was thinking about labels include tags in cloudwatch metricset. Here in ec2 metricset, the separator can be space or `|`, doesn't matter. It's only separating instanceID and metricName, like `i-1234` and `CPUUtilization`.